core/src/main/java/org/elasticsearch/index/query/HasParentQueryParser.java

```diff
@@ -42,16 +43,12 @@
 import java.util.Set;
 
 
-public class HasParentQueryParser extends BaseQueryParserTemp {
+public class HasParentQueryParser extends BaseQueryParser  {
 
+    private static final HasParentQueryBuilder PROTOTYPE = new HasParentQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
```

Comment:
I see that you put the prototype in the parser rather that in the builder. That is because it's the only place where it's used I guess (getBuilderPrototype)?  looks good